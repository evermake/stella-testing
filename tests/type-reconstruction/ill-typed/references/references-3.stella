language core;

extend with #general-recursion;
extend with #references;
extend with #type-reconstruction;

fn swap(p : auto) -> auto {
  return if p then p else *swap(true)
}

fn main(x : auto) -> auto {
  return swap(x)
}