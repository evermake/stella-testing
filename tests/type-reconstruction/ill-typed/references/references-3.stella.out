An error occurred during typechecking!
Type Error Tag: [ERROR_UNEXPECTED_TYPE_FOR_EXPRESSION]
when unifying expected type
  & Bool
against actual type
  Bool
when typechecking the expression
  swap (true)